api/utils.go

```diff
@@ -200,41 +190,41 @@ func countLeadingZeros(byteSlice []byte) (int, error) {
 	}
 	return 0, zeroInputError
 }
-func getTransactionsQuery(ctx context.Context, db *sql.DB, offset, limit int, chainid uint64, query string, params ...interface{}) ([]map[string]interface{}, error) {
+
+func getTransactions(ctx context.Context, db *sql.DB, offset, limit int, chainid uint64, whereClause string, includeTime bool, orderByTxIndex bool, params ...interface{}) ([]map[string]interface{}, error) {
+	var query string
+	selectFields := "blocks.hash, transactions.block, transactions.gas, transactions.gasPrice, transactions.hash, transactions.input, transactions.nonce, transactions.recipient, transactions.transactionIndex, transactions.value, transactions.v, transactions.r, transactions.s, transactions.sender, transactions.type, transactions.access_list, blocks.baseFee, transactions.gasFeeCap, transactions.gasTipCap, transactions.maxFeePerBlobGas, transactions.blobVersionedHashes, transactions.authListBytes"
+	if includeTime {
+		selectFields += ", blocks.time"
+	}
+	query = fmt.Sprintf("SELECT %v FROM transactions.transactions INNER JOIN blocks.blocks ON blocks.number = transactions.block WHERE transactions.rowid IN (SELECT transactions.rowid FROM transactions.transactions INNER JOIN blocks.blocks ON transactions.block = blocks.number WHERE %v) LIMIT ? OFFSET ?;", selectFields, whereClause)
+	if orderByTxIndex {
+		query = fmt.Sprintf("SELECT %v FROM transactions.transactions INNER JOIN blocks.blocks ON blocks.number = transactions.block WHERE %v ORDER BY transactions.transactionIndex LIMIT ? OFFSET ?;", selectFields, whereClause)
+	}
+
```

Comment:
Ok, so this is more complicated than we need I think. Looking back at the old implementation the legacy `getTransactions(` was actually never being used by the API. All of the singular tx methods were already using `getTransactionsBlock` so the `orderByTxIndex` select statement is not necessary. Also, instead of conditionally adding the time field I would just always include it in the select and the scan and conditionally apply it to the result.